Assets/Scripts/ScoreManager.cs

```diff
@@ -0,0 +1,53 @@
+using System;
+using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+
+public class ScoreManager : MonoBehaviour
+{
+    public int CurrentScore {  get; private set; }
+    public event Action OnScoreChanged;
+    private GameSettings gameSettings;
+    
+
+    public static ScoreManager instance;
+    public static ScoreManager Instance
+    {
+        get
+        {
+            if (instance == null)
+            {
+                GameObject go = new GameObject("ScoreManager");
+                instance = go.AddComponent<ScoreManager>();
+            }           
+            return instance;
+        }      
+    }
+    public void Initialize(GameSettings gameSettings)
+    {
+        instance.gameSettings = gameSettings;
+    }
+    
+    public IEnumerator AddPoints(List<GameObject> listOfMatches)
```

Comment:
I don't think this `AddPoints` should be an IEnumerator. Coroutines should be used when you need you have an iterative logic and you need to hold your application for each iteration to be done before moving forward. e.g you want to go over a list and do sth and you want to wait 1 sec between each element.
But here what you are doing is just incrementing a value and invoking an event. So let's change it to a normal function.

Reply:
Cool, thanks.